src/git/models/repository.ts

```diff
@@ -652,7 +652,8 @@ export class Repository implements Disposable {
 	}
 
 	private runTerminalCommand(command: string, ...args: any[]) {
-		runGitCommandInTerminal(command, args.join(' '), this.path, true);
+		const parsedArgs = args.map(arg => (arg.indexOf('#') >= 0 ? `"${arg}"` : arg));
```

Comment:
@bees4ever Does this need to be `>= 0` rather than just `=== 0`? Will the shell treat it as a comment even within a string?

Reply:
@eamodio sorry for the late replay, I saw you merged it now, wow :D I am very happy. Yes, string like "#my-branch-name" will be then _not_ interpreted as comment. I choose `>= 0` because a branch name *could* (I don't know why, but however) call `fix-#1234`.

Reply:
@bees4ever I tried it out with a branch like `fix-#1234` and it seemed to be fine without the quoting. Please let me know if you try it an see otherwise

Reply:
@eamodio Yes, correct, it works perfect, so however, my assumption was wrong. Anyway the bug is fixed. Thanks again.